### DISCUSSION TITLE
Shape of data for grouped/stacked bar chart

### DISCUSSION THREAD

**Discussion by dgwyer** [observablehq/plot#981] [Q&A] [answered]
I noticed that the _shape_ of the data for a [stacked bar chart](https://observablehq.com/@observablehq/plot-stack?collection=@observablehq/plot#cell-363) and [grouped bar chart](https://observablehq.com/@observablehq/plot-bar?collection=@observablehq/plot#cell-1005) shown in plot documentation is quite different.

Is it easily possible to use the same _shape_ of data for both charts to create a group and stacked bar chart to keep the data formatting needed to a minimum?

**Comment by Fil** [comment, marked as answer]
The shape of the data consumed by Plot is about the same in both examples; it's a "tidy" array with one element for each value, and fields that indicate the categories in which this value belongs.

in crimea: 
[{date: 1854-04-01, cause: "disease", deaths: 1}, ...];  // value is "deaths"

in stateage:
[{state: "AL", age: "<10", population: 598478}, ...] // value is "population"

what's different between the examples is the shape of the file that stores the data; the _stateage_ example shows how to reshape a "wide" file to extract it as tidy data.

**Reply by dgwyer** [reply]
I should have mentioned that I was referring to the shape of the data in the file rather than in the data consumed by Plot.

Thanks for the explanation of tidy vs wide data though. That makes it clearer.

**Reply by dgwyer** [reply]
I just did a test stacked and grouped bar chart example using the same data: https://observablehq.com/@dgwyer/grouped-and-stacked-bar-chart

But for the grouped chart I seem to have an issue with the facet axis labels on the x-axis. It doesn't seem to be formatted the same as the stacked chart label, which has the date formatted as `"narrow"`.

**Reply by Fil** [reply]
That's because the facets are rendered with the *fx* scale,  not the *x* scale.

**Reply by dgwyer** [reply]
Thanks, it's working now: https://observablehq.com/@dgwyer/grouped-and-stacked-bar-chart
